src/ServerRequestFactoryInterface.php

```diff
@@ -12,20 +12,9 @@
      *
      * @param string $method
      * @param UriInterface|string $uri
+     * @param array $server optional server params ($_SERVER or similar structure)
      *
      * @return ServerRequestInterface
      */
-    public function createServerRequest($method, $uri);
-
-    /**
-     * Create a new server request from server variables.
-     *
-     * @param array $server Typically $_SERVER or similar structure.
-     *
-     * @return ServerRequestInterface
-     *
-     * @throws \InvalidArgumentException
-     *  If no valid method or URI can be determined.
-     */
-    public function createServerRequestFromArray(array $server);
+    public function createServerRequest($method, $uri, $server = []);
```

Comment:
Yes, this is what I always defended: the server params is just an array of info that should not be used to generate the uri and method. These values must be calculated before and set explicity to the factory, in the same way than creating a `RequestInterface` instance. I also think that server params should be the latest argument in the factory, in order to be in sync with the RequestFactory and because the first two arguments (method and uri) are mandatory and the latest (server params) optional.